RHEL/6/input/checks/dhcp_client_restrict_options.xml

```diff
@@ -0,0 +1,139 @@
+<def-group>
+  <definition class="compliance" id="dhcp_client_restrict_options" version="1">
+    <metadata>
+      <title>Minimize DHCP Client Configured Options</title>
+      <affected family="unix">
+        <platform>Red Hat Enterprise Linux 6</platform>
+      </affected>
+      <description>Limit the options that the DHCP gets
+      and applies to the DHCP client.</description>
+      <reference source="galford" ref_id="20150317" ref_url="test_attestation" />
+    </metadata>
+    <criteria comment="DHCP is not being used or conditions are met"
+    operator="OR">
+      <extend_definition comment="dhcp package is not installed"
+      definition_ref="package_dhcp_removed" />
+      <criteria comment="Minimize DHCP client configured options"
+      operator="AND">
+        <criterion comment="Check domain-name in each dhclient .conf files"
+        test_ref="test_dhcp_client_domain-named" />
+        <criterion comment="Check domain-name-servers in each dhclient .conf files"
+        test_ref="test_dhcp_client_domain-name-servers" />
+        <criterion comment="Check nis-domain in each dhclient .conf files"
+        test_ref="test_dhcp_client_nis-domain" />
+        <criterion comment="Check nis-servers in each dhclient .conf files"
+        test_ref="test_dhcp_client_nis-servers" />
+        <criterion comment="Check ntp-servers in each dhclient .conf files"
+        test_ref="test_dhcp_client_ntp-servers" />
+        <criterion comment="Check routers in each dhclient .conf files"
+        test_ref="test_dhcp_client_routers" />
+        <criterion comment="Check time-offset in each dhclient .conf files"
+        test_ref="test_dhcp_client_time-offset" />
+        <criterion comment="Check subnet-mask in each dhclient .conf files"
+        test_ref="test_dhcp_client_subnet-mask" />
+        <criterion comment="Check broadcast-address in each dhclient .conf files"
+        test_ref="test_dhcp_client_broadcast-address" />
+      </criteria>
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="Checks the domain-name settings"
+  id="test_dhcp_client_domain-named" version="1">
+    <ind:object object_ref="obj_dhcp_client_domain-named" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_dhcp_client_domain-named"
+  version="1">
+    <ind:filepath>/etc/dhcp/dhclient*.conf</ind:filepath>
```

Comment:
Unless I have overlooked something pretty obvious, this filepath won't work for other config file than exactly the ```/etc/dhcp/dhclient\*.conf``` one. Default element operation, when not specified is the ```equals``` one. So this check will search for exactly ```/etc/dhcp/dhclient\*.conf``` location.

But instead of that I suppose the OVAL check is trying to check any dhclient*.conf file under /etc/dhcp directory (IOW e.g. ```/etc/dhcp/dhclient-sample.conf``` should pass the check too under assumption it's configured correctly).

Here's the current output of this check running on RHEL-6 with the setting applied as specified in the XCCDF prose:
```
# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 6.6 (Santiago)

# cat /etc/dhcp/dhclient-sample.conf 
supersede domain-name "example.com";
supersede domain-name-servers 192.168.1.2;
supersede nis-domain "";
supersede nis-servers "";
supersede ntp-servers "ntp.example.com ";
supersede routers 192.168.1.1;
supersede time-offset -18000;
request subnet-mask;
require subnet-mask;
supersede broadcast-address 192.168.1.1;
```
OpenSCAP collects the following data on that system (selected only relevant parts for brevity):
```
# oscap xccdf eval --profile C2S_tailored --tailoring-file /root/dhcp.xml --oval-results /usr/share/xml/scap/ssg/content/ssg-rhel6-xccdf.xml 
Title   Minimize the DHCP-Configured Options
Rule    dhcp_client_restrict_options
Result  fail
```
Inspecting ```ssg-rhel6-oval.xml.result.xml``` then unveils:
```
...
<collected_objects>
          <object id="oval:ssg:obj:1814" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1815" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1816" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1817" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1818" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1819" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1820" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1821" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:1822" version="1" flag="does not exist"/>
          <object id="oval:ssg:obj:2322" version="1" flag="complete">
            <reference item_ref="1030801"/>
          </object>
        </collected_objects>
...
```
Moving the ```/etc/dhcp/dhclient-sample.conf``` to be ```/etc/dhcp/dhclient\*.conf``` lets the test to collect the objects && pass:
```
# mv /etc/dhcp/dhclient-sample.conf /etc/dhcp/dhclient\*.conf
# oscap xccdf eval --profile C2S_tailored --tailoring-file /root/dhcp.xml --oval-results /usr/share/xml/scap/ssg/content/ssg-rhel6-xccdf.xml 
Title   Minimize the DHCP-Configured Options
Rule    dhcp_client_restrict_options
Result  pass
```
This behaviour was tested on:
```
# rpm -q openscap
openscap-1.0.8-1.el6_5.1.x86_64
```
So instead of:
```
<ind:filepath>/etc/dhcp/dhclient*.conf</ind:filepath>
```

IMHO the check should be doing something like:
```
<ind-def:filepath operation="pattern match">/etc/dhcp/dhclient.*\.conf</ind-def:filepath>
```
(or even possibly escape all the the forward slashes with backlashes in the regex too like:
```
<ind:filepath operation="pattern match">\/etc\/dhcp\/dhclient.*\.conf</ind:filepath>
```
)

This way when the settings is also in the ```/etc/dhcp/dhclient-sample.conf``` conf, the check will scan it.

Same change should be applied also to the other ```<filepath>``` occurrences in this OVAL check below.